.github/workflows/integration.yaml

```
@@ -0,0 +1,30 @@
+name: Integration workflow
+
+on:
+  push:
+    branches: ['main']
+  pull_request:
+    branches: ['main']
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [16.x, 18.x, 22.x]
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node-version }}
+          cache: 'npm'
+
+      - name: Install dependencies
+        run: npm install
+
+      - name: Run server
+        run: npm run dev
```

Add a `npm run build` command to your package.json that transpiles the code
```suggestion
       - name: Install dependencies
        run: npm ci

      - name: Check for linting issues
        run: npm run lint

      - name: Build dist
        run: npm run build --if-present
```